hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/ChangeDetectionPolicy.java

```diff
@@ -277,11 +295,32 @@ public String getRevisionId(ObjectMetadata objectMetadata, String uri) {
       return objectMetadata.getETag();
     }
 
+    @Override
+    public String getRevisionId(S3ObjectAttributes s3Attributes) {
+      return s3Attributes.getETag();
+    }
+
+    @Override
+    public String getRevisionId(CopyResult copyResult) {
+      return copyResult.getETag();
+    }
+
     @Override
     public void applyRevisionConstraint(GetObjectRequest request,
         String revisionId) {
-      LOG.debug("Restricting request to etag {}", revisionId);
-      request.withMatchingETagConstraint(revisionId);
+      if (revisionId != null) {
+        LOG.debug("Restricting request to etag {}", revisionId);
```

Comment:
could clarify "get request" / "copy request" in these log messages so that they are unique